### PR TITLE
MGMT-2030 - Add Red Hat IT Root CA to ignition and InstallConfig

### PR DESCRIPTION
**Pull request by rollandf**
In order to be able to pull container images from Brew, the root certificate for Red Hat IT needs to be installed on the live ISO.
Also for assisted-install-controller, the cert is added to InstallConfig in `AdditionalTrustBundle`

### CODE REVIEW

**Comment by rollandf** [comment]
@eranco74 @gamli75 @ori-amizur

**Comment by rollandf** [comment]
```
Sep 06 11:59:52 localhost systemd[1]: Starting agent.service...
Sep 06 11:59:53 test-infra-cluster-master-1 podman[1289]: Trying to pull registry-proxy.engineering.redhat.com/rh-osbs/assisted-installer-assisted-installer-agent:v4.6.0-5...
Sep 06 11:59:54 test-infra-cluster-master-1 podman[1289]: Getting image source signatures
Sep 06 11:59:54 test-infra-cluster-master-1 podman[1289]: Copying blob sha256:47db82df7f3f4393c1f19c362a2db2c47ca049b6fb20bef041dfc9bdb12a4504
Sep 06 11:59:54 test-infra-cluster-master-1 podman[1289]: Copying blob sha256:abfb08018cf591e4fdd6fbdf3110ffb13e9e65779d0c7cecd98c2c46038dd0a1
Sep 06 11:59:54 test-infra-cluster-master-1 podman[1289]: Copying blob sha256:fa172450bf3acc2be8bd538632a26f64f4fd70328206458fde08381266f9f9fc
Sep 06 11:59:54 test-infra-cluster-master-1 podman[1289]: Copying blob sha256:77c58f19bd6e67185938abb6bbb6ec229e07a5e607453904294d982de141d2f0
Sep 06 11:59:57 test-infra-cluster-master-1 podman[1289]: Copying config sha256:e3c505358b754f88d390bcd98e2a5038c7b5ea28c52d0a6f86e9031e27c4dc92
Sep 06 11:59:57 test-infra-cluster-master-1 podman[1289]: Writing manifest to image destination
Sep 06 11:59:57 test-infra-cluster-master-1 podman[1289]: Storing signatures
S
```

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -121,7 +149,16 @@ const ignitionConfigFormat = `{
           "name": "root"
       },
       "contents": { "source": "data:,{{.AGENT_MOTD}}" }
-    }]
+	},
+	{
+	  "overwrite": true,
+	  "path": "/etc/pki/ca-trust/source/anchors/rh-it-root-ca.crt",
```
Comment (rollandf) [inline internal/bminventory/inventory.go:155]: There is a Systemd process on boot that will update the CA from the directory `/etc/pki/ca-trust/source/anchors` :
See https://github.com/coreos/fedora-coreos-config/pull/120

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -95,6 +95,34 @@ sudo journalctl -u agent.service
 **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  ** **  **  **  **  **  **  **
 `
 
+const redhatRootCA = `
```
Comment (rollandf) [inline internal/bminventory/inventory.go:98]: Expires on 2055
```
 $ openssl x509 -in RH-IT-Root-CA.crt -text -noout
......
        Validity
            Not Before: Jul  6 17:38:11 2015 GMT
            Not After : Jun 26 17:38:11 2055 GMT
```

**Comment by romfreiman** [comment]
Do we want to have it by default? Send it to customers as well? Maybe we should protect it with some flag and use it only in the lab?
What sounds like a something that can be generic enough - is to parse the images in the config-map - and if u see brew path there, you can append it to the ignition.
what do you think?

**Comment by rollandf** [comment]
> Do we want to have it by default? Send it to customers as well? Maybe we should protect it with some flag and use it only in the lab?
> What sounds like a something that can be generic enough - is to parse the images in the config-map - and if u see brew path there, you can append it to the ignition.
> what do you think?

@avishayt @ronniel1 what do you think?

**Comment by ronniel1** [comment]
> > Do we want to have it by default? Send it to customers as well? Maybe we should protect it with some flag and use it only in the lab?
> > What sounds like a something that can be generic enough - is to parse the images in the config-map - and if u see brew path there, you can append it to the ignition.
> > what do you think?
> 
> @avishayt @ronniel1 what do you think?

I think that we should add it only on our lab.
I don't think we should give these certificates to customers

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -95,6 +95,34 @@ sudo journalctl -u agent.service
...
+-----BEGIN CERTIFICATE-----
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:99]: I don't think the certificate should be in github, for everyone to see

Comment (rollandf) [reply internal/bminventory/inventory.go:99]: It is available from https://password.corp.redhat.com/RH-IT-Root-CA.crt without VPN

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -95,6 +96,34 @@ sudo journalctl -u agent.service
 **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  **  ** **  **  **  **  **  **  **
 `
 
+const redhatRootCA = `
```
Comment (YuviGold) [inline internal/bminventory/inventory.go:99]: Can it be saved in a file?

Comment (rollandf) [reply internal/bminventory/inventory.go:99]: Why is it better?

File: deploy/assisted-service.yaml [hunk 5]
```diff
@@ -102,6 +102,8 @@ spec:
               value: "false"
             - name: LOG_LEVEL
               value: "info"
+            - name: INSTALL_RH_CA
+              value: "false"
```
Comment (eranco74) [inline deploy/assisted-service.yaml:106]: When will it be set to true?

Comment (rollandf) [reply deploy/assisted-service.yaml:106]: When the QE will need to test images from brew

File: internal/installcfg/installcfg.go [hunk 6]
```diff
@@ -235,6 +236,9 @@ func GetInstallConfig(log logrus.FieldLogger, cluster *common.Cluster) ([]byte,
 	if err != nil {
 		return nil, err
 	}
+	if addRhCa {
+		cfg.AdditionalTrustBundle = fmt.Sprintf(` | %s`, ca)
```
Comment (romfreiman) [inline internal/installcfg/installcfg.go:240]: @rollandf can u open a task on @carbonin to sypport append of some fields, and then also adjust this code?

**Comment by romfreiman** [comment]
/approve
/lgtm
(assuming it works)

**Comment by rollandf** [comment]
/hold

Subsytem failing on:
[Fail] cluster install - DHCP install cluster cases [It] Install with DHCP

**Comment by carbonin** [comment]
I wonder if this is something that should be set up outside of the source in whatever automation QE is using to run their tests.
I already added customization for the install config and I should have the ignition side soon.

**Comment by carbonin** [comment]
For example, it would be a great addition to assisted-test-infra to call the install config and ignition config patch endpoints to add this.

**Pull request merged**